src/main/java/com/haxwell/apps/quizki/services/QuestionServiceImpl.java

```diff
@@ -212,5 +214,37 @@ public CreatedQuestionDTO createQuestion(CreateQuestionDTO cqDTO) throws CreateQ
 		return outputDTO;
 	}
 	
+	public CreatedQuestionDTO getQuestionById(String ids) throws GetQuestionException {
+		
+		long id = Long.parseLong(ids);
+		
+		outQuestion = questionRepo.findById(id);
+		
+		if(outQuestion.isPresent()) {
+			
+			outputDTO = new CreatedQuestionDTO();
+			question = outQuestion.get();
+			
+			outputDTO.setId(question.getId());
+			outputDTO.setUserId(question.getUser().getId());
+			outputDTO.setDescription(question.getDescription());
+			outputDTO.setText(question.getText());
+			outputDTO.setDifficulty(question.getDifficulty());
+			outputDTO.setQuestionType(question.getQuestionType());
+			outputDTO.setTopics(question.getTopics());
+			outputDTO.setReferences(question.getReferences());
+			outputDTO.setChoices(question.getChoices());
+			
+			
+		} else {
+			
+			ValidationErrorData data = new ValidationErrorData();
+			data.addFieldError("id", "question.id.not.found");
```

Comment:
I've seen that the framework generated error messages have this format. The ones that are the result of validation errors for example. I was emulating that format thinking that if the frontend has to parse them they would be consistent. We could make our error messages different but would that cause problems? Not sure how they would be handling them on the frontend.

Reply:
Can you point me at these "framework generated errors"? Before I say "No, it won't cause problems." I want to be sure you and I have the same thing in mind.

Reply:
Sure, if you run the `nullNameDTO` Postman test it send a bad request that is detected by the built in framework annotation `@NotNull` which triggers an exception and a handler that is part of the framework. Or in other words the framework code is creating and handling this error and you get this...

`{
    "errors": [
        {
            "field": "name",
            "message": "NotBlank.userCreateDTO.name"
        }
    ]
}`

The other built in `@Validations` work the same way like `@Email` , `@Size` etc.
I have been modeling the messages I put in the `ValidationErrorData` for all of the messages generated by the "non-validation" exceptions in this same period delimited format. Although I think that I have not followed the same `validation.class.field` format in all cases in an attempt to make them a bit more readable.

Reply:
The `message` is designed to be a human readable explanation of an error. The front end is not depending on the `message` field to be in a certain format. What we need, actually, is a numeric ID on the error object, so that the specific error can be determined. That is TBD, I suppose.

But I agree with my earlier assessment: the string messages in the error objects should not be period delimited. They should appear as normal sentences.

Reply:
I'll remove them in these files. I was trying to change the messages created by the `@Validations` when I was figuring out how to create custom exceptions and did not get it to work, but I'm pretty sure it can be done. I suggest that before I work on messages besides the ones I generate in this PR we come up with a consistent format and I'll go back and rewrite the rest on a separate issue, if that makes sense?